sdk/python/lib/test/automation/test_config_file.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2024-2024, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+from io import StringIO
+import unittest
+
+from pulumi.automation import LocalWorkspace, Stack, ProjectSettings
+from pulumi.automation._stack import _parse_extra_args
+from .test_utils import stack_namer
+
+
+class TestConfigFile(unittest.IsolatedAsyncioTestCase):
+    async def test_config_file_option(self):
+        """Tests that the config_file option is correctly passed to the operations"""
+        config_file_path = os.path.join(
+            os.path.dirname(__file__), "data", "yaml", "Pulumi.dev.yaml"
+        )
+
+        def pulumi_program():
+            from pulumi import Config, export
+
+            config = Config()
+            export("plain", config.get("plain"))
+
+        try:
+            project_name = "test_project"
+            stack_name = "dev"
+
+            workspace = LocalWorkspace(
+                work_dir=tempfile.mkdtemp(),
+                project_settings=ProjectSettings(name=project_name, runtime="python"),
+                program=pulumi_program,
+            )
+
+            stack = Stack.create_or_select(stack_name, workspace)
+
+            preview_args = _parse_extra_args(config_file=config_file_path)
+            self.assertEqual(stack.preview(preview_args).change_summary.items.same, 1)
```

Comment:
doesn't `stack.preview()` just need the `config_file=config_file_path` argument, rather than passing the parsed args?  Same for the other `stack.*` commands.

Reply:
probably, I assumed `_parse_extra_args()` was a helper that properly constructed them. Maybe not, should I swap this back, or leave these with the other old asserts also

Reply:
No, any function starting with an underscore in python is a private function (by convention), so end users should never call this.